syntax ="proto3";


message receta {
    string titulo = 1;
    string descripcion=2;
    string pasos=3;
    uint32 tiempoEnMinutos=4;
    string categoria=5;
    string creador=6;
    string foto1=7;
    string foto2=8;
    string foto3=9;
    string foto4=10;
    string foto5=11;
    uint32 idReceta = 12;
}

message ingredienteDeReceta {
    string nombre = 1;
    uint32 cantidad = 2;
    string tipoDeMedida =3;
}

message listaIngredientesPorResponse  {
    repeated ingredienteDeReceta ingredientes = 1;
}

message traerRecetasPorRequest {
    string tiempoEnMinutosMIN = 1;
    string tiempoEnMinutosMAX = 2;
    string categoria = 3;
    string creador = 4;
    string titulo=5;
    string ingrediente = 6;
}
message crearRecetaRequest {
    string titulo = 1;
    string descripcion=2;
    string pasos=3;
    uint32 tiempoEnMinutos=4;
    string categoria=5;
    string creador=6;
    string foto1=7;
    string foto2=8;
    string foto3=9;
    string foto4=10;
    string foto5=11;
}
message status {
    // 0 = error, 1 = ok
   int32 status=1;
}
message traerRecetasPorResponse {
    repeated receta recetas = 1;
}

message agregarRecetaAFavoritosRequest{
    string usuario = 1;
    string idReceta = 2;
}

message agregarRecetaAFavoritosResponse{
    string mensaje = 1;
    /*
        mensajes:
            "Agregado a ingredientes"
            "Error"
    */
}

message agregarIngredienteARecetaRequest {
    string idReceta = 1;
    string ingrediente = 2;
    uint32 cantidad = 3;
    string tipoDeMedida = 4;
}

message agregarIngredienteARecetaResponse{
    string mensaje = 1;
    /*
        mensajes:
            "Agregado a favoritos"
            "Error"
    */
}

message traerRecetasFavoritasRequest{
    string usuario = 1;
}

message traerRecetaRequest{
    string idReceta = 1;
}

service servicioReceta{
    rpc traerRecetasPor (traerRecetasPorRequest) returns (traerRecetasPorResponse);
    rpc crearReceta(crearRecetaRequest)returns (status);
    rpc agregarRecetaAFavoritos(agregarRecetaAFavoritosRequest) returns (agregarRecetaAFavoritosResponse);
    rpc traerRecetasFavoritas(traerRecetasFavoritasRequest) returns (traerRecetasPorResponse);
    rpc traerReceta (traerRecetaRequest) returns (receta);
    rpc traerIngredientesDeReceta (traerRecetaRequest) returns (listaIngredientesPorResponse);
    rpc agregarIngredienteAReceta(agregarIngredienteARecetaRequest) returns (agregarIngredienteARecetaResponse);
}